pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/EventLogController.java

```diff
@@ -0,0 +1,92 @@
+import java.util.Date;
+import java.util.ResourceBundle;
+
+import net.sourceforge.pmd.util.fxdesigner.model.LogEntry;
+import net.sourceforge.pmd.util.fxdesigner.model.LogEntry.Category;
+
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.TableCell;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableColumn.SortType;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.cell.PropertyValueFactory;
+
+/**
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class EventLogController implements Initializable {
+
+    private final DesignerApp designerApp;
+    private final MainDesignerController parent;
+
+    @FXML
+    private TableView<LogEntry> eventLogTableView;
+    @FXML
+    private TableColumn<LogEntry, Date> logDateColumn;
+    @FXML
+    private TableColumn<LogEntry, Category> logCategoryColumn;
+    @FXML
+    private TableColumn<LogEntry, String> logMessageColumn;
+    @FXML
+    private TextArea logDetailsTextArea;
+
+
+    public EventLogController(DesignerApp owner, MainDesignerController mainController) {
+        this.designerApp = owner;
+        parent = mainController;
+    }
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        logCategoryColumn.setCellValueFactory(new PropertyValueFactory<>("category"));
+        logMessageColumn.setCellValueFactory(new PropertyValueFactory<>("message"));
+        final DateFormat dateFormat = new SimpleDateFormat("HH:mm:ss");
+        logDateColumn.setCellValueFactory(
+            entry -> new SimpleObjectProperty<>(entry.getValue().getTimestamp()));
+        logDateColumn.setCellFactory(column -> new TableCell<LogEntry, Date>() {
+            @Override
+            protected void updateItem(Date item, boolean empty) {
+                super.updateItem(item, empty);
+                if (item == null || empty) {
+                    setText(null);
+                    setGraphic(null);
+                } else {
+                    setText(dateFormat.format(item));
+                }
+            }
+        });
+
+        eventLogTableView.setItems(designerApp.getLogger().getLog());
+
+        eventLogTableView
+            .getSelectionModel()
+            .selectedItemProperty()
+            .addListener((obs, oldVal, newVal) -> logDetailsTextArea.setText(
+                newVal == null ? "" : newVal.getStackTrace()));
+
+        eventLogTableView.resizeColumn(logMessageColumn, -1);
+
+
+        logMessageColumn.prefWidthProperty()
+                        .bind(eventLogTableView.widthProperty()
+                                               .subtract(logCategoryColumn.getPrefWidth())
+                                               .subtract(logDateColumn.getPrefWidth())
+                                               .subtract(2)); // makes it work
```

Comment:
By default the table doesn't visually fit the pane it's given. This binding allows resizing of the splitpane, the only thing is that the border of the table isn't visible without this 2 pixel adjustment. It might just be on my machine, and I might just be picky